cloud-aws-common/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/common/AwsInstanceConnector.java

```diff
@@ -151,10 +156,15 @@ private List<CloudVmInstanceStatus> setCloudVmInstanceStatuses(AuthenticatedCont
             Long timeboundInMs) {
         AmazonEc2Client amazonEC2Client = new AuthenticatedContextView(ac).getAmazonEC2Client();
         try {
-            Collection<String> instances = instanceIdsWhichAreNotInCorrectState(vms, amazonEC2Client, status);
+            ImmutablePair<Collection<String>, Collection<String>> instancesPair = instanceIdsWhichAreNotInCorrectState(vms, amazonEC2Client, status);
+            Collection<String> instances = instancesPair.left;
+            Collection<String> pendingInstances = instancesPair.right;
```

Comment:
Same here: Using an `ImmutablePair` to determine pending instances can be refactored. What happens if an additional unfavourable state needs to be added here?

Reply:
Made a new set on instancestatus just like the completed ones so in future if any more instance have to be added that it can be added there.